File: src/www/__test__/macros/MatcherMacro.js
import AllMatchers from "../matchers/AllMatchers";
import AbstractRegExpMacro from "./AbstractRegExpMacro";

export default class MatcherMacro extends AbstractRegExpMacro {
  constructor() {
    super(/^\s+[^\s]/);

    this._allMatchers = new AllMatchers();
  }

  async interpretMatch(match, instruction, context) {
    if (context.matcherPaused) return;

    await this._allMatchers.interpretInstruction(instruction, context);
  }
}

File: src/test/java/com/drpicox/game/macros/MatcherMacro.java
package com.drpicox.game.macros;

import com.drpicox.game.interpreter.Context;
import com.drpicox.game.interpreter.Instruction;
import com.drpicox.game.matchers.AllMatchers;
import org.springframework.stereotype.Component;

import java.util.regex.MatchResult;

@Component
public class MatcherMacro extends AbstractRegExpMatcher {

    private AllMatchers allMatchers;

    public MatcherMacro(AllMatchers allMatchers) {
        super("\\s+[^\\s].*");
        this.allMatchers = allMatchers;
    }

    @Override
    protected void interpretMatch(MatchResult match, Instruction instruction, Context context) {
        if (context.has("matcherPaused")) return;
        allMatchers.interpretInstruction(instruction, context);
    }
}
